diff --git a/CMakeLists.txt b/CMakeLists.txt
index 90899a4..883f92e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -114,26 +114,15 @@ if (USE_CRYPTO STREQUAL "OpenSSL")
 		if(WIN32 AND OPENSSL_USE_STATIC_LIBS)
 			list(APPEND CMAKE_REQUIRED_LIBRARIES ws2_32 crypt32)
 		endif()
-		check_symbol_exists(EVP_MD_CTX_free openssl/evp.h OPENSSL_NEW_ENOUGH)
-		if (NOT OPENSSL_NEW_ENOUGH)
-			message(FATAL_ERROR "Cannot find EVP_MD_CTX_free in OpenSSL headers/libs for the target architecture.  Check that you're using OpenSSL 1.1.0 or later.")
-		endif()
 	cmake_pop_check_state()
 	cmake_push_check_state()
 		set(CMAKE_REQUIRED_LIBRARIES OpenSSL::Crypto)
 		if(WIN32 AND OPENSSL_USE_STATIC_LIBS)
 			list(APPEND CMAKE_REQUIRED_LIBRARIES ws2_32 crypt32)
 		endif()
-		if(USE_CRYPTO25519 STREQUAL "OpenSSL")
-			check_symbol_exists(EVP_PKEY_get_raw_public_key openssl/evp.h OPENSSL_HAS_25519_RAW)
-		endif()
 	cmake_pop_check_state()
 endif()
 
-if(USE_CRYPTO25519 STREQUAL "OpenSSL" AND NOT OPENSSL_HAS_25519_RAW)
-	message(FATAL_ERROR "Cannot find (EVP_PKEY_get_raw_public_key in OpenSSL headers/libs for the target architecture.  Please use -DUSE_CRYPTO25519=Reference or upgrade OpenSSL to 1.1.1 or later")
-endif()
-
 if(USE_CRYPTO STREQUAL "libsodium" OR USE_CRYPTO25519 STREQUAL "libsodium")
 	find_package(sodium REQUIRED)
 endif()
