diff --git a/libshaderc/CMakeLists.txt b/libshaderc/CMakeLists.txt
index 5fd58ef..c86f469 100644
--- a/libshaderc/CMakeLists.txt
+++ b/libshaderc/CMakeLists.txt
@@ -75,10 +75,10 @@ endif(SHADERC_ENABLE_INSTALL)
 
 find_package(Threads)
 set(SHADERC_LIBS
-  glslang OSDependent OGLCompiler glslang ${CMAKE_THREAD_LIBS_INIT}
+  glslang::glslang glslang::OSDependent glslang::OGLCompiler glslang::glslang ${CMAKE_THREAD_LIBS_INIT}
   shaderc_util
-  SPIRV # from glslang
-  SPIRV-Tools
+  glslang::SPIRV # from glslang
+  SPIRV-Tools-static
 )
 
 if(NOT BUILD_SHARED_LIBS)
@@ -105,7 +105,7 @@ shaderc_add_tests(
 
 shaderc_add_tests(
   TEST_PREFIX shaderc_shared
-  LINK_LIBS shaderc_shared SPIRV-Tools
+  LINK_LIBS shaderc_shared SPIRV-Tools-shared
   INCLUDE_DIRS include ${shaderc_SOURCE_DIR}/libshaderc_util/include ${glslang_SOURCE_DIR}
                ${spirv-tools_SOURCE_DIR}/include
                ${SPIRV-Headers_SOURCE_DIR}/include
diff --git a/third_party/CMakeLists.txt b/third_party/CMakeLists.txt
index 7bd9617..22deb99 100644
--- a/third_party/CMakeLists.txt
+++ b/third_party/CMakeLists.txt
@@ -41,7 +41,7 @@ if (IS_DIRECTORY ${SHADERC_SPIRV_HEADERS_DIR})
   add_subdirectory(${SHADERC_SPIRV_HEADERS_DIR} spirv-headers)
 endif()
 
-if (NOT TARGET SPIRV-Tools)
+if (NOT TARGET SPIRV-Tools-static)
   # Check SPIRV-Tools before glslang so that it is linked into glslang.
   # we control optimizations via glslang API calls directly.
   if (IS_DIRECTORY ${SHADERC_SPIRV_TOOLS_DIR})
@@ -58,7 +58,9 @@ if (NOT TARGET SPIRV-Tools)
     endif()
     add_subdirectory(${SHADERC_SPIRV_TOOLS_DIR} spirv-tools)
   endif()
-  if (NOT TARGET SPIRV-Tools)
+  find_package(SPIRV-Tools CONFIG REQUIRED GLOBAL)
+  find_package(SPIRV-Tools-opt CONFIG REQUIRED GLOBAL)
+  if (NOT TARGET SPIRV-Tools-static)
     message(FATAL_ERROR "SPIRV-Tools was not found - required for compilation")
   endif()
 endif()
